data/config-schema/src/main/resources/net/corda/schema/configuration/db/1.0/corda.db.json

```diff
@@ -51,6 +51,19 @@
               "type": "integer",
               "minimum": 1,
               "default": 10
+            },
+            "min_size": {
+              "description": "The minimal database pool size.",
+              "default": null,
+              "anyOf": [
+                {
+                  "type": "integer",
+                  "minimum": 0
+                },
+                {
+                  "type": "null"
```

Comment:
Why do we want this as optional parameter? Why not have it as not nullable with a defaulted value?

Reply:
Not really sure why it should be the maximum though? Maybe we should be adding a check to assert that minumValue <= maximumValue but that should be it?

Reply:
That's what Hikari [recommends:](https://github.com/brettwooldridge/HikariCP) "... for maximum performance and responsiveness to spike demands, we recommend not setting this value and instead allowing HikariCP to act as a fixed size connection pool. Default: same as maximumPoolSize". I think they know what they are talking about.

Reply:
Our use case is slightly different to the typical use case of a web application that operates under a predictable workload and wants a fully initialised and always warm pool. I think actually a default of min pool size = 0 might be a more sensible default.

If we set the default of max pool size to 10 for example (looking at the config this seems to already be the case), then in a cluster that has 10 vnodes deployed, we'll be reserving up 100 connections by default even though we might need to use a much smaller number of them. This can be an issue in clusters that want to run in a cost-efficient way and with relatively small footprint at the database level.

Reply:
This configuration is for the virtual node and the cluster databases...

For the node pools, the min size will be zero, so we will not reserve any connection (and we will evacuate any unused connection after 2 minutes).

Reply:
[Discussed a bit offline with Yiftach]

We currently have a single schema (and as a result a single default) for vnode and cluster DBs. That's a bit confusing and maybe something we want to change at some point in the future (not something for this PR), since each case has possibly different requirements. For example, for the cluster DB having min pool size = max pool size might make sense as a default, but for vnode DBs it probably doesn't as explained above.  

Now from a practical point of view, the associated runtime-os [PR](https://github.com/corda/corda-runtime-os/pull/3909) overrides the defaults for vnode DBs to set min pool size to 0. And it sets min pool size = max pool size for cluster DB connections, which is the current behaviour (and the default in the config). So, I think that's ok for now.